rfc_do_not_merge/RFC.md

```diff
@@ -0,0 +1,268 @@
+# FeatureToggle Component
+
+## Todo
+
+- [ ] What is a feature Flag
+
+- [ ] When is it needed ?
+
+- [ ] What it is not (security)
```

Comment:
A feature is not necessarily related to a user. It could be the current date or an environment variable.

The feature can be checked before the user is retrieved. It could be useful when you want to disable access to a feature without checking credentials.

But the FeatureTogglee can be used to check handle permissions! You can use this component to control different kind of toggles, even coupling them (eg: envvar + user)!

In a more advanced way, a feature can be enabled for some randomly chosen users (canary testing). In this case, the Security component is not enough. In the future, the FeatureToggle component could handle this using the Cache component.